services/board/src/Domain/MovingStrategies/PerSquareCount.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace CliChess\Board\Domain\MovingStrategies;
+
+use CliChess\Board\Domain\Square;
+
+class PerSquareCount implements Moving
+{
+    public function __construct(
+        private readonly int $count
+    ) {
+    }
+
+    public function canMove(Square $from, Square $to): bool
+    {
+        return $this->count === abs($from->rowDiff($to)) || $this->count === abs($from->columnDiff($to));
```

Comment:
What do you think about adding the methods for abs diff at `Square`? Maybe we could call absolute diff as `distance`: `rowDistance` and `columnDistance`.